server/src/main/java/io/druid/segment/realtime/appenderator/FiniteAppenderatorDriver.java

```diff
@@ -178,47 +185,80 @@ public void clear() throws InterruptedException
   }
 
   /**
-   * Add a row. Must not be called concurrently from multiple threads.
+   * Add a row.  This method may internally incur persisting data added so far.  Also, if too large data are persisted,
```

Comment:
@pjain1 @dclim does the above sound useful to you too, especially in the Kafka context? I'm wondering if it can reduce some of the complexity from #4178, maybe making DriverHolders unnecessary since a single AppenderatorDriver could handle multiple handoffs.